homeassistant/components/ld2410_ble/coordinator.py

```diff
@@ -26,15 +32,43 @@ def __init__(self, hass: HomeAssistant, ld2410_ble: LD2410BLE) -> None:
         ld2410_ble.register_callback(self._async_handle_update)
         ld2410_ble.register_disconnected_callback(self._async_handle_disconnect)
         self.connected = False
+        self._last_update_time = NEVER_TIME
+        self._debounce_cancel: CALLBACK_TYPE | None = None
+        self._debounced_update_job = HassJob(
+            self._async_handle_debounced_update,
+            f"LD2410 {ld2410_ble.address} BLE debounced update",
+        )
+
+    @callback
+    def _async_handle_debounced_update(self, _now: datetime) -> None:
+        """Handle debounced update."""
+        self._debounce_cancel = None
+        self._last_update_time = time.monotonic()
+        self.async_set_updated_data(None)
 
     @callback
     def _async_handle_update(self, state: LD2410BLEState) -> None:
         """Just trigger the callbacks."""
         self.connected = True
-        self.async_set_updated_data(None)
+        previous_last_updated_time = self._last_update_time
+        self._last_update_time = time.monotonic()
+        if self._last_update_time - previous_last_updated_time >= DEBOUNCE_SECONDS:
+            self.async_set_updated_data(None)
+            return
+        if self._debounce_cancel is None:
+            self._debounce_cancel = async_call_later(
```

Comment:
What's the reason we don't use our debounce helper?

Reply:
It comes in via a callback and creating tasks added more complexity than the call later

Reply:
Ok. 👍